src/main/java/net/fabricmc/tinyremapper/extension/mixin/common/ResolveUtility.java

```diff
@@ -76,7 +76,7 @@ public ResolveUtility(TrEnvironment environment, Logger logger) {
 		Comparator<T> comparator;
 
 		if ((flag & FLAG_NON_SYN) != 0) {
-			comparator = (x, y) -> Boolean.compare(x.isSynthetic(), y.isSynthetic()) == 0
+			comparator = (x, y) -> Boolean.compare(x.isSynthetic(), y.isSynthetic()) != 0
```

Comment:
You can use `Comparator.comparing(TrMember::isSynthetic).thenComparing(Comparator.comparing(TrMember::getIndex))`